hail/src/main/scala/is/hail/backend/service/ServiceBackend.scala

```diff
@@ -254,7 +263,9 @@ class ServiceBackend(
     ReferenceGenome.getReference(name).toJSONString
   }
 
-  private[this] def execute(ctx: ExecuteContext, _x: IR): Option[(Annotation, PType)] = {
+  private[this] def execute(ctx: ExecuteContext, _x: IR, bufferSpecString: String): Array[Byte] = {
+    // FIXME: do we need TypeCheck(ctx, _x)?
```

Comment:
When you `Compile` on line 272, that calls `TypeCheck`, so I think you're ok without `TypeCheck`

Reply:
And Validate?